soccer/gameplay/plays/offense/distraction.py

```diff
@@ -0,0 +1,178 @@
+import main
+import robocup
+import behavior
+import constants
+import enum
+import standard_play
+import evaluation
+import tactics.coordinated_pass
+import skills.move
+import skills.capture
+
+
+class distraction(standard_play.StandardPlay):
+
+    class State (enum.Enum):
+        setup = 1, 'capture ball and move distractor and striker into position'
+        optional_adjustment = 2, 'setup a pass to the center right'
+        cetner_pass = 3, 'pass to the center right of the field'
+        passing = 4, 'pass from first capture to distractor'
+        cross = 5, 'pass from distractor to striker'
+        shoot = 6, 'shoot at goal'
+    
+    def __init__(self):
+        super().__init__(continuous=False)  
+
+        self.ball_is_far = False
+        self.distraction_point = robocup.Point(0.40*constants.Field.Width, 0.95*constants.Field.Length) #the first distraction point
+        self.distraction_recieve_pass_point = robocup.Point(0.40*constants.Field.Width, 0.8*constants.Field.Length) #the second distraction point
+        self.striker_point = robocup.Point(-0.40*constants.Field.Width, 0.9*constants.Field.Length) #striker's position
+        self.center = robocup.Point(0.5*constants.Field.Width,0.5*constants.Field.Length) #center of field position, used if ball is far
+        self.distract_box_min = robocup.Point(0.33*constants.Field.Width/2,(7/9)*constants.Field.Length)
+        self.distract_box_max = robocup.Point(constants.Field.Width/2, constants.Field.Length)
+        self.striker_box_min = robocup.Point(-0.33*constants.Field.Width/2,(7/9)*constants.Field.Length)
+        self.striker_box_max = robocup.Point(-constants.Field.Width/2, constants.Field.Length)
```

Comment:
These can be robocup.Rect's. I can't remember if it's Rect or Rectangle, but yeah

Reply:
This is how you create a rect

https://github.com/RoboJackets/robocup-software/blob/f0cf06638ee427563ca7162bbbd65809c30f59f2/soccer/gameplay/tactics/positions/defender.py#L35